src/libPMacc/include/random/distributions/Uniform_float.h

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright 2015 Alexander Grund
+ *
+ * This file is part of libPMacc.
+ *
+ * libPMacc is free software: you can redistribute it and/or modify
+ * it under the terms of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * libPMacc is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with libPMacc.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "types.h"
+
+namespace PMacc {
+namespace random {
+namespace distributions {
+
+    /**
+     * Returns a random float value uniformly distributed in [0,1)
+     */
+    template<class T_RNGMethod = bmpl::_1>
+    class Uniform_float
```

Comment:
If we add the result type as first parameter we can remove the double implementation `Uniform_uint32` and call this `Uniform32bit`.
If a special implementation for a type like `int32` is needed a specialization can be implemented.